src/main/java/com/faforever/client/chat/CategoryOrChatUserListItem.java

```diff
@@ -13,6 +13,15 @@
  */
 @Value
 class CategoryOrChatUserListItem {
+
   ChatUserCategory category;
   ChatChannelUser user;
+
+  public boolean isUserItem() {
+    return user != null;
+  }
+
+  public boolean isCategoryItem() {
+    return category != null;
+  }
```

Comment:
These are only ever used followed by an .equals().
Wouldn't it be easier to write a
`public boolean contains(Object item)` ?

Reply:
Ok, it is a good idea.

Reply:
I will not write this method. There is an easier way.

Reply:
See please https://github.com/FAForever/downlords-faf-client/pull/2022/files#diff-55c0277ffe170fd8988215bc63e3d03909588a2d72fb0fa2e9cca785cdd12bb9R500

Reply:
Why did you drop `@Value`? You could just configure the default constructor to be private and call the private constructor from your custom . If you drop @Value you loose ToString, Equals and HashCode features from Lombok

Reply:
Reverted.